scripts/plot_performance_over_commit_range.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
+
+# Given a range of benchmark jsons created by run_benchmarks_over_commit_range.sh, plots a graph showing the
+# performance development over time
+
+import matplotlib.pyplot as plt
+import os
+import pandas as pd
+from pydriller import RepositoryMining
+import re
+import sys
+
+if(len(sys.argv) != 4):
+    print("Usage: " + sys.argv[0] + " <path to Hyrise> <first commit> <last commit>")
+    exit()
+
+results = []
+column_names = ['Commit']
+
+for commit in RepositoryMining(sys.argv[1], from_commit=sys.argv[2], to_commit=sys.argv[3]).traverse_commits():
+	result = []
+	result.append(commit.hash[:6] + " " + commit.msg.partition('\n')[0])
+
+	json_path = "auto_" + commit.hash + ".json"
+
+	if not os.path.exists(json_path):
+		print("JSON file not found: " + json_path)
+		for column in column_names:
+			result.append(None)
+		continue
+
+	with open(json_path) as file:
+		# If column headers are unset, set them
+		if len(column_names) == 1:
+			for name in re.findall(r'name": "(.*)"', file.read()):
+				column_names.append(name)
+
+			file.seek(0)
+
+		# Quick and dirty way to retrieve the throughput. This is "better" than parsing the JSON because while the JSON has
+		# changed over time, this field has been available for a long time.
+		for items_per_second in re.findall(r'items_per_second": ([0-9]+\.[0-9]*)', file.read()):
```

Comment:
Is this something we should have a binary/systemTest (or whatever they are called) for?

Reply:
I would consider that overkill. If this plot script doesn't work anymore, it is sufficient to fix it once it becomes a problem.

Reply:
Yes, it sounds like an overkill. But isn't the overhead pretty big once we realize after 15 commits that some changes broke this script?

Reply:
So what if it's broken? Fixing it will take the same amount of time no matter when we notice it.